### PR TITLE
Bp 2642 add response and push validation

### PR DESCRIPTION
**Pull request by Rinor12010**

### CODE REVIEW

File: src/Handlers/Reply/ReplyHandler.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            } catch (e){
+                let objData = {}
+                new URLSearchParams(data).forEach((value, name)=>{
+                    objData[name] = value
+                })
+                this.data = objData
+            }
+        }else {
+            this.data = data
+        }
```
Comment (ShuCh3n) [inline src/Handlers/Reply/ReplyHandler.ts:27]: What other data type can you expect other than a string?

File: src/Handlers/Reply/ReplyHandler.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        let hmac = new Hmac(HttpMethods.POST,this.uri,this.data,nonce,time)
+
+        let hash = hmac.hashData(hmac.getHashString())
+
+        return crypto.timingSafeEqual(Buffer.from(hash),Buffer.from(providedHash))
+    }
+    private validateHttp(data:object){
+        let brq_signature = data['brq_signature']
+        let stringData = ''
+        delete data['brq_signature']
```
Comment (ShuCh3n) [inline src/Handlers/Reply/ReplyHandler.ts:60]: "brq_signature" is not always lowercase, in plaza you are able to change the settings to upper, lower and mixed cases.

File: src/Handlers/Reply/ReplyHandler.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    validate() {
+        if(this.data["Key"] && this.auth_header && this.uri){
+            this._isValid = this.validateJson(this.auth_header)
+        }else if (this.data["brq_signature"]){
+            this._isValid = this.validateHttp({...this.data})
+        }else {
+            throw new Error('Invalid reply data')
+        }
+        return this
+    }
```
Comment (ShuCh3n) [inline src/Handlers/Reply/ReplyHandler.ts:44]: Please try to limit your if-else statement

```javascript
validate() {
        if(this.data["Key"] && this.auth_header && this.uri) {
            this._isValid = this.validateJson(this.auth_header)
return this
        }

if (this.data["brq_signature"]){
            this._isValid = this.validateHttp({...this.data})
return this
        }
       
 throw new Error('Invalid reply data')
    }
```

This way is much easier to read.

**Review by ShuCh3n — CHANGES_REQUESTED** [review]
Minor changes

File: src/Handlers/Reply/ReplyHandler.ts (hunk 1 above)
Comment (Rinor12010) [reply src/Handlers/Reply/ReplyHandler.ts]: Was thinking of cases where the data is already parsed so there is no need to parse it again

**Pull request closed without merging**